server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultConnectOptionsContext.java

```diff
@@ -208,6 +210,19 @@ private Integer getHttpKeepaliveTimeout() {
         return httpKeepaliveTimeout;
     }
 
+    private Integer getHttpKeepaliveMaxConnections() {
+        int maxConnections = DEFAULT_HTTP_KEEPALIVE_MAX_CONNECTIONS;
+        String connectionsValue = options.get("http.keepalive.max.connections");
+        if (connectionsValue != null) {
+            int val = Integer.parseInt(connectionsValue);
+            if (val > 0) {
+                maxConnections = val;
```

Comment:
It is possible to disable keep-alive using <http.keepalive>disabled</http.keepalive> connect option. It may be done for security reasons. It is also useful if there is a bug in reusing cached connections or if we know that upstream server doesn't support keepalive etc.

But it may be useful to just warn if the number is negative and use the default value.